src/util/type_byte_size.cpp

```diff
@@ -469,26 +469,17 @@ const expr2tc &get_base_object(const expr2tc &expr)
   return expr;
 }
 
-/* XXX fbrausse: what does this function compute? */
+// Obtain the format string (the first argument) in scanf and printf function
 const irep_idt get_string_argument(const expr2tc &expr)
```

Comment:
If this function did what the comment says, then how can this function also be used also on the parameters following the format string? This is done in two places: https://github.com/esbmc/esbmc/blob/69fe5a70b54dc8a177547bc504b83de83422fce0/src/goto-programs/goto_check.cpp#L345 and https://github.com/esbmc/esbmc/blob/69fe5a70b54dc8a177547bc504b83de83422fce0/src/goto-symex/builtin_functions.cpp#L364

Reply:
Okay, I made a mistake here. 

Previously, this function is used to get the fmt_string. And Now it is extended to dereference and get every argument. E.g.

```c++
scanf("%d %d",&x, &y);
fmt_string = get_string_argument(func_call.operands[0]);  // "%d %d"
arg_1 = get_string_argument(func_call.operands[1]);  // x
arg_2 = get_string_argument(func_call.operands[2]);  // y
```

Reply:
Please see the discussion with @mikhailramalho below. There are fundamental problems using this function if it can return the constant-string's value *or* the symbol's id. Example:
```c
void f(const char *fmt) // call me with "%s" as the only format directive
{
printf(fmt, "string literal");
// the format "string" is now a symbol and the parameter is now a constant-string
}
```

My suggestion: now that you've implemented it ontop of get_base_object(), it might be worth removing this function altogether and directly using get_base_object() instead of it. Then you can also properly distinguish between symbol expressions and constant expressions.